ReactAndroid/src/main/java/com/facebook/react/uimanager/animation/NativeAnimatedNodesManager.java

```diff
@@ -0,0 +1,344 @@
+    if (parentNode == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + parentNodeTag +
+        " does not exists");
+    }
+    AnimatedNode childNode = mAnimatedNodes.get(childNodeTag);
+    if (childNode == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + childNodeTag +
+        " does not exists");
+    }
+    parentNode.addChild(childNode);
+  }
+
+  public void disconnectAnimatedNodes(int parentNodeTag, int childNodeTag) {
+    AnimatedNode parentNode = mAnimatedNodes.get(parentNodeTag);
+    if (parentNode == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + parentNodeTag +
+        " does not exists");
+    }
+    AnimatedNode childNode = mAnimatedNodes.get(childNodeTag);
+    if (childNode == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + childNodeTag +
+        " does not exists");
+    }
+    parentNode.removeChild(childNode);
+  }
+
+  public void connectAnimatedNodeToView(int animatedNodeTag, int viewTag) {
+    AnimatedNode node = mAnimatedNodes.get(animatedNodeTag);
+    if (node == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + animatedNodeTag +
+        " does not exists");
+    }
+    if (!(node instanceof PropsAnimatedNode)) {
+      throw new JSApplicationIllegalArgumentException("Animated node connected to view should be" +
+        "of type " + PropsAnimatedNode.class.getName());
+    }
+    PropsAnimatedNode propsAnimatedNode = (PropsAnimatedNode) node;
+    if (propsAnimatedNode.mConnectedViewTag != -1) {
+      throw new JSApplicationIllegalArgumentException("ANimated node " + animatedNodeTag + " is " +
+        "already attached to a view");
+    }
+    propsAnimatedNode.mConnectedViewTag = viewTag;
+  }
+
+  public void disconnectAnimatedNodeFromView(int animatedNodeTag, int viewTag) {
+    AnimatedNode node = mAnimatedNodes.get(animatedNodeTag);
+    if (node == null) {
+      throw new JSApplicationIllegalArgumentException("Animated node with tag " + animatedNodeTag +
+        " does not exists");
+    }
+    if (!(node instanceof PropsAnimatedNode)) {
+      throw new JSApplicationIllegalArgumentException("Animated node connected to view should be" +
+        "of type " + PropsAnimatedNode.class.getName());
+    }
+    PropsAnimatedNode propsAnimatedNode = (PropsAnimatedNode) node;
+    if (propsAnimatedNode.mConnectedViewTag == viewTag) {
+      propsAnimatedNode.mConnectedViewTag = -1;
+    }
+  }
+
+  /**
+   * Animation loop performs two DFSes over the graph of animated nodes. We use incremented
+   * {@code mAnimatedGraphDFSColor} to mark nodes as visited in each of the DFSes which saves
+   * additional loops for clearing "visited" states.
+   *
+   * First DFS starts with nodes that are in {@code mUpdatedNodes} or directly attached to an active
```

Comment:
Comment what it means for a node to be in mUpdatedNodes. It's not obvious to me so I'm going to have to go read the code to find out